### PR TITLE
feat: recaptcha demosite - update control flow from client

### PR DESCRIPTION
**Pull request by Sita04**
Update control flow from the client to create_assessment call.
New flow: Client calls the respective post action method which inturn calls the create assessment.

Example: client -> on_signup -> create_assessment

### CODE REVIEW

**Review by faube — CHANGES_REQUESTED** [review]

File: recaptcha_enterprise/demosite/app/static/scripts/global.js [hunk 1]
```diff
@@ -12,34 +12,100 @@
...
 
-function createAssessment({ action, token }) {
-  // SEE: Code for fetching the assessment from backend goes here
-  // SEE: Refer to demo app backend code for more information
-  // SEE: If using a library or framework, can fetch its usual way
-  return fetchDemoAssessment({ action, token });
+function homepage({token}) {
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      action : RECAPTCHA_ACTIONS.home,
```
[the remark this answers is not in the document]
Comment (Sita04) [reply recaptcha_enterprise/demosite/app/static/scripts/global.js]: Yes, thanks for the catch! Removed it.

**Review by faube — APPROVED** [review]

File: recaptcha_enterprise/demosite/app/static/scripts/global.js [hunk 2]
```diff
@@ -12,47 +12,120 @@
...
 
-function useAssessment(score) {
-  // SEE: Code for handling the assessment goes here
-  showAssessmentInDemo(score);
+async function signup({token}) {
+  // Do not pass plain text credentials here. Always encrypt and follow security standards.
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+      username: "",
```
Comment (rsamborski) [inline recaptcha_enterprise/demosite/app/static/scripts/global.js:43]: Shouldn't we pass the actual username and password from the form?

File: recaptcha_enterprise/demosite/app/static/scripts/global.js [hunk 3]
```diff
@@ -12,47 +12,120 @@
...
+  });
+  useAssessment(score);
+}
+
+async function login({token}) {
+  // Do not pass plain text credentials here. Always encrypt and follow security standards.
+  const json_body = JSON.stringify({
+    recaptcha_cred: {
+      token: token,
+      username: "",
```
Comment (rsamborski) [inline recaptcha_enterprise/demosite/app/static/scripts/global.js:60]: ditto

File: recaptcha_enterprise/demosite/app/static/scripts/global.js [hunk 4]
```diff
@@ -12,47 +12,120 @@
...
 
-function fetchDemoAssessment({ action, token }) {
-  const body = JSON.stringify({
+async function comment({token}) {
+  const json_body = JSON.stringify({
     recaptcha_cred: {
-      action,
-      token,
+      token: token,
+      comment: "",
```
Comment (rsamborski) [inline recaptcha_enterprise/demosite/app/static/scripts/global.js:91]: Shouldn't we pass the actual comment from the form?

**Review by rsamborski — APPROVED** [review]
Approved with couple questions.

File: recaptcha_enterprise/demosite/app/static/scripts/global.js (hunk 4 above)
Comment (Sita04) [reply recaptcha_enterprise/demosite/app/static/scripts/global.js:91]: ditto

File: recaptcha_enterprise/demosite/app/static/scripts/global.js (hunk 2 above)
Comment (rsamborski) [reply recaptcha_enterprise/demosite/app/static/scripts/global.js:43]: Thanks. Sounds reasonable and a comment mentioning that this should be used to pass user provided data would be very helpful.

Comment (Sita04) [reply recaptcha_enterprise/demosite/app/static/scripts/global.js:44]: Thanks. Added the comment.

**Pull request merged**